safekeeper/src/timelines_global_map.rs

```diff
@@ -296,8 +296,8 @@ impl GlobalTimelines {
         global_lock
             .timelines
             .values()
-            .cloned()
             .filter(|t| !t.is_cancelled())
+            .cloned()
```

Comment:
FYI the idea here was to `clone` as soon as possible, to release `TIMELINES_STATE` lock sooner. But now I realize that lock is probably still held until the end of the function, so this change LGTM.

Reply:
Yeah 2 reasons it's held open for the entire duration:
1. cloned() is still lazy on iterators, so it's till referencing the locked contents while collecting
2. temporaries (like mutex guards) remain until the end of the entire statement, and only drop then

Reply:
so to summarize, with the changes the lock is freed sooner even, as less cloning happens, so the whole iterator is done sooner.